src/pages/peoplePage.jsx

```diff
@@ -1,12 +1,112 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { connect } from "react-redux";
+import { useHistory } from "react-router-dom";
+import * as peopleAction from "../actions/peopleAction";
+import { Typography, List, Card, Divider, Skeleton, notification } from "antd";
+import InfiniteScroll from "react-infinite-scroll-component";
+const { Title } = Typography;
+const { Meta } = Card;
 
 const People = (props) => {
+  const { userToken, userList, getAllUsers } = props;
+  const [data, setData] = useState([]);
+  const [dataIndex, setDataIndex] = useState(0);
+  const history = useHistory();
+  const itemAmount = 12;
 
-  return (
-    <div className="row-container">
-        People Page
-    </div>
-  );
+  useEffect(() => {
+    if (userToken) {
+      getAllUsers(userToken);
+    } else {
+      notification['error']({
+        message: 'Please Login',
+        description:
+          'Sorry. You don\'t have authorization to People Page. Please login or sign up.',
+      });
+      history.push("/");
+    }
+  }, [userToken, getAllUsers, history]);
+  useEffect(() => {
+    if (userList && userList?.length > 0) {
+      loadMoreData();
+    }
+  }, [userList]);
+
+  const loadMoreData = () => {
+    if (userList && userList?.length > 0) {
+      if (dataIndex + itemAmount > userList.length - 1) {
+        setData([...data, ...userList.slice(dataIndex)]);
+        setDataIndex(userList.length - 1);
+      } else {
+        setData([...data, ...userList.slice(dataIndex, dataIndex + itemAmount)]);
+        setDataIndex(dataIndex + itemAmount);
+      }
+    }
+  };
+
+  if (!userList && userList?.length === 0) {
+    return;
+  } else {
+    return (
+      <div className="people-page-backgroup">
+        <Title level={2}>All people</Title>
+        <InfiniteScroll
```

Comment:
You might want to hide this component while it's still fetching the data, otherwise, it might break the infinite scroll.

```suggestion
        {loaded && <InfiniteScroll
```